packages/frontend/web-ui/src/common/hooks/useShowPassword.ts

```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react';
+
+export const useShowPassword = (initial: boolean) => {
+  const [showPassword, setShowPassword] = useState<boolean>(initial);
+
+  const handleClickShowPassword = () => {
+    setShowPassword((show: boolean) => !show);
+  };
+
+  const handleMouseDownPassword = (
+    event: React.FormEvent<HTMLInputElement>,
```

Comment:
We don't really care which element is being targeted (btw, it's a button, not an input), so `Event<HTMLElement>` is way more convenient

Reply:
I've seen another type to named this element, you can use it for any event: **_e: React.SyntheticEvent_**

Reply:
`React.MouseEvent` is probably more convenient then

Reply:
Done!